cmd/exporter.go

```diff
@@ -79,21 +82,52 @@ func main() {
 	flag.Parse()
 
 	klog.Infof("Kepler running on version: %s", kversion.Version)
+
 	if *cpuProfile != "" {
 		f, err := os.Create(*cpuProfile)
-		if err == nil {
-			profErr := pprof.StartCPUProfile(f)
-			if profErr == nil {
-				sigs := make(chan os.Signal, 1)
-				signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
-				go func() {
-					<-sigs
-					fmt.Println("exiting...")
-					pprof.StopCPUProfile()
-					os.Exit(0)
-				}()
+		if err != nil {
+			klog.Fatal("could not create CPU profile: ", err)
+		}
+		if err := pprof.StartCPUProfile(f); err != nil {
+			klog.Fatal("could not start CPU profile: ", err)
+		}
+		klog.Infof("Started CPU profiling")
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+		go func() {
+			defer f.Close()
+			select {
+			case <-sigs:
+				break
```

Comment:
if SIGTERM raised, also exit kepler program

Reply:
thanks @rootfs, I updated it